subprojects/resources-http/src/main/java/org/gradle/internal/resource/transport/http/WebdavUtils.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.resource.transport.http;
+
+public class WebdavUtils {
+
+    public static String stripPrefix(String url) {
+        return url.replaceFirst("^dav\\+", "");
+    }
```

Comment:
I'm confused, if your just strippifg of the dav prefix off the URL, why do you need a custom uploader? Just use a HTTP url instead?

Reply:
The dav+ prefix is just to indicate that WebDAV is required. I'm not an expert on WebDAV, but what I've observed is that the regular HTTP uploader doesn't work with a WebDAV server because each directory must be explicitly created with the correct permissions.